extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTableWriter.java

```diff
@@ -75,191 +77,181 @@ public class IcebergTableWriter {
+    private final Schema userSchema;
 
     /**
-     * Reverse mapping from Iceberg field IDs to Deephaven column names.
+     * Mapping from Iceberg field IDs to Deephaven column names.
      */
-    private final Map<Integer, String> icebergFieldIdToDhColumn;
+    private final Map<Integer, String> fieldIdToColumnName;
 
     /**
-     * The schema to use when in conjunction with the {@link #dhColumnsToIcebergFieldIds} to map Deephaven columns from
-     * {@link #tableDefinition} to Iceberg columns.
+     * Initialized lazily from the {@value TableProperties#DEFAULT_NAME_MAPPING} property in the table metadata, if
+     * needed.
      */
-    private final Schema userSchema;
-
-    @Nullable
-    private final TableMetadata tableMetadata;
     @Nullable
     private Map<String, Integer> nameMappingDefault;
 
-    private Schema schemaToWrite;
-    private PartitionSpec partitionSpecToWrite;
+    /**
+     * The factory to create new output file locations for writing data files.
+     */
+    private final OutputFileFactory outputFileFactory;
 
-    private OutputFileFactory outputFileFactory;
 
     IcebergTableWriter(
             final TableWriterOptions tableWriterOptions,
             final IcebergTableAdapter tableAdapter) {
-        this.tableDefinition = tableWriterOptions.tableDefinition();
-        this.dhColumnsToIcebergFieldIds = tableWriterOptions.dhColumnsToIcebergFieldIds();
-        this.icebergFieldIdToDhColumn = new HashMap<>();
-        this.table = tableAdapter.icebergTable();
-        this.userSchema = tableWriterOptions.schema().orElseGet(table::schema);
-        verifyFieldIds(dhColumnsToIcebergFieldIds.values(), userSchema);
         this.tableAdapter = tableAdapter;
+        this.table = tableAdapter.icebergTable();
 
         if (table instanceof HasTableOperations) {
             tableMetadata = ((HasTableOperations) table).operations().current();
         } else {
             tableMetadata = null;
         }
-        inferSpecAndSchemaToWrite();
-    }
 
-    /**
-     * Check that all the field IDs are present in the schema.
-     */
-    private void verifyFieldIds(final Collection<Integer> fieldIds, final Schema schema) {
-        if (!fieldIds.isEmpty()) {
-            for (final Integer fieldId : fieldIds) {
-                if (schema.findField(fieldId) == null) {
-                    throw new IllegalArgumentException("Column corresponding to field ID " + fieldId + " not " +
-                            "found in schema, available columns in schema are: " + schema.columns());
-                }
+        this.tableDefinition = tableWriterOptions.tableDefinition();
+
+        // We only write non-partitioning columns to parquet file
+        this.parquetTableDefinition = TableDefinition.of(tableDefinition.getColumnStream()
+                .filter(columnDefinition -> !columnDefinition.isPartitioning())
+                .collect(Collectors.toList()));
```

Comment:
We determined that the spec dictates that we write partitioning columns, right?

Reply:
Yes, making that change now.